### PR TITLE
Add `mypy` to the project

### PR DESCRIPTION
**Pull request by Viicos**

### CODE REVIEW

File: src/otpauth/core.py [hunk 1]
```diff
@@ -2,34 +2,37 @@
 import typing as t
 from urllib.parse import quote
 from abc import ABCMeta, abstractmethod
+from .types import SupportedAlgorithms
 
+Self = t.TypeVar("Self", bound="OTP")
```
Comment (Viicos) [inline src/otpauth/core.py:7]: If you want to use `typing_extensions.Self` instead, I can make use of it

File: src/otpauth/core.py [hunk 2]
```diff
@@ -64,19 +67,18 @@ def string_code(self, code: int) -> str:
             >>> otp.string_code(123)
             '000123'
 
-        :param code: The number that this OTP generated.
+        :param code: The number that this OTPnb generated.
```
Comment (lepture) [inline src/otpauth/core.py:70]: typo?  what is `OTPnb`?

Comment (Viicos) [reply src/otpauth/core.py:70]: Indeed a typo

Comment (Viicos) [reply src/otpauth/core.py:70]:
```suggestion
        :param code: The number that this OTP generated.
```

File: src/otpauth/types.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import typing as t
+
+SupportedAlgorithms = t.Literal["SHA1", "SHA256", "SHA512"]
```
Comment (lepture) [inline src/otpauth/types.py:3]: How about putting this type into `core.py`, so that we don't have to add a new file.

Comment (Viicos) [reply src/otpauth/types.py:3]: Did this at first, but created a new file as I'm using it as well in the other files. I can revert if you prefer.

**Pull request merged**